token/service.go

```diff
@@ -57,10 +57,7 @@ func (s *tokenService) ResolveTargetToken(ctx context.Context, target, token str
 	if err != nil {
 		return "", "", errors.Wrapf(err, "error while decoding the token for %s", target)
 	}
-	if externalToken.Username == nil {
```

Comment:
?

Reply:
The Auth service changed its API contract to require Username on output.  Because of this, Goa changed the generated type from a pointer-to-string to an actual string.  Because it's an actual string (so never nil) and required (so never length 0) I took out the test and removed a later indirection.

If you wish I can add a test to ensure non-empty string; I just felt that was redundant.

Reply:
Re-enable the check for empty string. We want to know specifically if it's "" instead of some random unknown failure later. Can't trust @alexeykazakov ;)

Reply:
Unless @alexeykazakov can confirm it's always there now. :)

Reply:
It's always there unless we have a bug in Auth :) So, don't trust me :)

Reply:
To be clear. Username should be always there. It's why we changed it from optional to required. But trust no one! Auth can have bugs (hard to believe though). Check the output and report an error if it's not what you expected.

Reply:
Test added.  Best part is the previous code had a bug in it: missing %s !